src/storage/storage_structure/storage_structure.cpp

```diff
@@ -63,6 +63,72 @@ void BaseColumnOrList::readBySequentialCopy(Transaction* transaction,
     }
 }
 
+void BaseColumnOrList::readRelIDsBySequentialCopy(Transaction* transaction,
+    const shared_ptr<ValueVector>& vector, PageElementCursor& cursor,
+    const std::function<page_idx_t(page_idx_t)>& logicalToPhysicalPageMapper,
+    table_id_t commonTableID, bool hasNoNullGuarantee) {
+    uint64_t numValuesToRead = vector->state->originalSize;
+    uint64_t vectorPos = 0;
+    auto setVectorNullFunc = getSetVectorNullFunc(hasNoNullGuarantee);
+    while (vectorPos != numValuesToRead) {
+        uint64_t numValuesInPage = numElementsPerPage - cursor.elemPosInPage;
+        uint64_t numValuesToReadInPage = min(numValuesInPage, numValuesToRead - vectorPos);
+        auto physicalPageIdx = logicalToPhysicalPageMapper(cursor.pageIdx);
+        readRelIDsFromAPageBySequentialCopy(transaction, vector, vectorPos, physicalPageIdx,
+            cursor.elemPosInPage, numValuesToReadInPage, commonTableID, setVectorNullFunc);
+        vectorPos += numValuesToReadInPage;
+        cursor.nextPage();
+    }
+}
+
+void BaseColumnOrList::readRelIDsFromAPageBySequentialCopy(Transaction* transaction,
```

Comment:
Let's discuss this, relID and nodeID should be scanned with the same code path ideally. I understand right now we can nodeID compression which prevents you from using the same logic. However, if eventually we can have a unified design, we should open an issue to keep track of this.

Ditto for readRelIDsBySequentialCopy and readRelIDsBySequentialCopyWithSelState